subprojects/docs/src/docs/userguide/commandLineInterface.adoc

```diff
@@ -0,0 +1,520 @@
+
+image::img/console-animation.gif[alt="Gradle Rich Console",data-alt="foo.gif",class="js-gif-playback-control gif-control gif-control__paused"]
+
+Features:
+
+ * Logs above grouped by task that generated them
+ * Progress bar and timer visually describe overall status
+ * Parallel work-in-progress lines below describe what is happening now
+
+=== Execution Options
+The following options affect how builds are executed, by changing what is built or how dependencies are resolved.
+
+`-a`, `--no-rebuild` (deprecated)::
+Do not rebuild project dependencies.
+
+`--include-build`::
+Run the build as a composite, including the specified build. See <<composite_builds, Composite Builds>>.
+
+`--offline`::
+Specifies that the build should operate without accessing network resources. Learn more about <<cache_command_line_options,options to override dependency caching>>.
+
+`--refresh-dependencies`::
+Refresh the state of dependencies. Learn more about how to use this in the dependency management docs.<<cache_command_line_options, dependency management docs>>.
+
+`--dry-run`::
+Run Gradle with all task actions disabled. Use this to show which task would have executed.
+
+[[sec:command_line_bootstrapping_projects]]
+=== Bootstrapping New Projects
+
+==== Creating new Gradle builds
+Use the built-in `gradle init` task to create a new Gradle builds, with new or existing projects.
+
+----
+❯ gradle init
+----
+
+Most of the time you'll want to specify a project type. Available types include `basic` (default), `java-library`, `java-application`, and more. See <<build_init_plugin, init plugin documentation>> for details.
+
+----
+❯ gradle init --type java-library
+----
+
+==== Standardize and provision Gradle
+The built-in `gradle wrapper` task generates a script, `gradlew`, that invokes a declared version of Gradle, downloading it beforehand if necessary.
+
+----
+❯ gradle wrapper --gradle-version=4.4
+----
+
+You can also specify `--distribution-type=(bin|all)`, `--gradle-distribution-url`, `--gradle-distribution-sha256-sum` in addition to `--gradle-version`. Full details on how to use these options are documented in the <<gradle_wrapper,Gradle wrapper section>>.
+
+[[sec:command_line_debugging]]
+=== Debugging Options
+
+`-?`, `-h`, `--help`::
+Shows a help message with all available CLI options.
+
+`-v`, `--version`::
+Prints Gradle, Groovy, Ant, JVM, and operating system version information.
+
+`-S`, `--full-stacktrace`::
+Print out the full (very verbose) stacktrace for any exceptions. See also <<sec:command_line_logging, logging options>>.
+
+`-s`, `--stacktrace`::
+Print out the stacktrace also for user exceptions (e.g. compile error).  See also <<sec:command_line_logging, logging options>>.
+
+`--scan`::
+Create a https://gradle.com/build-scans[build scan] with fine-grained information about all aspects of your Gradle build.
+
+`-Dorg.gradle.debug=true`::
+Debug Gradle client (non-Daemon) process.
```

Comment:
:x: Should mention which remote debug port a user can connect with.

Reply:
Yep. Good catch.

Reply:
done